### PR TITLE
Fix definition of ITEM_NOT_FOUND for pre-1.13 versions.

### PR DESCRIPTION
**Pull request by jkeenan**
Signed-off-by: James E Keenan <jkeenan@cpan.org>

### CODE REVIEW

File: ext/GDBM_File/GDBM_File.xs
Comment (steve-m-hay) [inline ext/GDBM_File/GDBM_File.xs]: Typos: Should say "...functions would..."

File: ext/GDBM_File/GDBM_File.xs
Comment (steve-m-hay) [inline ext/GDBM_File/GDBM_File.xs]: Typo: Should say "...existed..."

File: ext/GDBM_File/GDBM_File.xs
Comment (steve-m-hay) [inline ext/GDBM_File/GDBM_File.xs:149]: Are there versions earlier than 1.0 that are likely to be in use? E.g. What if someone is using 0.9?...

**Review by steve-m-hay — COMMENTED** [review]
I don't know anything about GDBM so I can't really say whether the change is correct, but it looks reasonable aside from the minor comments I've added.

File: ext/GDBM_File/GDBM_File.xs
Comment (jkeenan) [reply ext/GDBM_File/GDBM_File.xs]: Done in next commit to branch; also next typo.

File: ext/GDBM_File/GDBM_File.xs
Comment (jkeenan) [reply ext/GDBM_File/GDBM_File.xs:149]: The code and the comment were originally composed by one of the two GNU project maintainers of GDBM (whom I've contacted off-list re this ticket).  In addition, releases with a `0` major version, if they ever existed, are no longer available for download at the [GNU site](http://ftp.gnu.org/gnu/gdbm/).  So I think we're safe here.

Comment (jkeenan) [reply ext/GDBM_File/GDBM_File.xs:149]: Corrected typos; corrected $VERSION; squashed commits and force-pushed smoke-me branch.

**Comment by Leont** [comment]
> I don't know anything about GDBM so I can't really say whether the change is correct, but it looks reasonable aside from the minor comments I've added.

Same, @graygnuorg is the person to ask

**Review by Leont — APPROVED** [review]

**Comment by Leont** [comment]
It fixes the problem on my machine

**Comment by jkeenan** [comment]
> It fixes the problem on my machine

Thanks.  Could you cast a vote for this in the maint-votes branch?

@steve-m-hay, the commit listed in the maint-votes branch will have some merge conflicts and need a `$VERSION` bump.  I did this in this pull request, but I'll leave it up to you to figure out the best way to apply it to maint-5.34 and the next RC tarball.

**Comment by graygnuorg** [comment]
Yes, this should fix the bug. GDBM_* are enum constants, not defines, so that #ifdef is not able to catch them.

**Comment by steve-m-hay** [comment]
On Sat, 5 Mar 2022 at 00:25, James E Keenan ***@***.***>
wrote:

> It fixes the problem on my machine
>
> Thanks. Could you cast a vote for this in the maint-votes branch?
>
> @steve-m-hay <https://github.com/steve-m-hay>, the commit listed in the
> maint-votes branch will have some merge conflicts and need a $VERSION
> bump. I did this in this pull request, but I'll leave it up to you to
> figure out the best way to apply it to maint-5.34 and the next RC tarball.
>
Now merged. I took the commit before it too (
aacd2398e766500cb5d83c4d76b642fcf31d997a) as the simplest way to avoid
merge conflicts. I also took  the commit after it (
9fa00226c78c14bc34a562b90346047d2dcdd3b1), which fixes the same spelling
mistakes that I spotted in review! I hadn't realized they'd already been
fixed in blead.

**Pull request closed without merging**
